packages/camera/lib/camera.dart

```diff
@@ -0,0 +1,278 @@
+        : new Container();
+  }
+}
+
+class CameraValue {
+  /// True if the camera is on.
+  final bool isStarted;
+  final bool initialized;
+  final String errorDescription;
+
+  const CameraValue({this.isStarted, this.initialized, this.errorDescription});
+
+  const CameraValue.uninitialized()
+      : this(isStarted: true, initialized: false);
+
+  bool get isErroneous => errorDescription != null;
+
+  CameraValue copyWith({
+    bool isStarted,
+    bool initialized,
+    String errorDescription,
+  }) {
+    return new CameraValue(
+      isStarted: isStarted ?? this.isStarted,
+      initialized: initialized ?? this.initialized,
+      errorDescription: errorDescription ?? this.errorDescription,
+    );
+  }
+
+  @override
+  String toString() {
+    return '$runtimeType('
+        'started: $isStarted, '
+        'initialized: $initialized, '
+        'errorDescription: $errorDescription)';
+  }
+}
+
+/// Controls a device camera.
+///
+/// Use [availableCameras] to get a list of available cameras.
+///
+/// Before using a [CameraController] a call to [initialize] must complete.
+///
+/// To show the camera preview on the screen use a [CameraPreview] widget.
+class CameraController extends ValueNotifier<CameraValue> {
+  final CameraDescription description;
+  final ResolutionPreset resolutionPreset;
+  int _textureId;
+  bool _disposed = false;
+  StreamSubscription<Map<String, dynamic>> _eventSubscription;
+  Completer<Null> _creatingCompleter;
+  Size _previewSize;
+
+  CameraController(this.description, this.resolutionPreset)
+      : super(const CameraValue.uninitialized());
+
+  /// The size of the preview in pixels.
+  ///
+  /// Is [null] until [initialize] completes.
+  Size get previewSize => _previewSize;
+
+  /// Convenience getter for `previewSize.height / previewSize.width`.
+  ///
+  /// Can only be called after [initialize] completes
+  double get aspectRatio => previewSize.height / previewSize.width;
+
+  /// Initializes the camera on the device.
+  ///
+  /// Throws a [CameraException] if the initialization fails.
+  Future<Null> initialize() async {
+    if (_disposed) {
+      return;
+    }
+    try {
+      _creatingCompleter = new Completer<Null>();
+      final Map<String, dynamic> reply = await _channel.invokeMethod(
+        'create',
+        <String, dynamic>{
+          'cameraName': description.name,
+          'resolutionPreset': serializeResolutionPreset(resolutionPreset),
+        },
+      );
+      _textureId = reply['textureId'];
+      _previewSize = new Size(
+        reply['previewWidth'].toDouble(),
+        reply['previewHeight'].toDouble(),
+      );
+      value = value.copyWith(initialized: true);
+      _applyStartStop();
+    } on PlatformException catch (e) {
+      value = value.copyWith(errorDescription: e.message);
+      throw new CameraException(e.code, e.message);
+    }
+    _eventSubscription =
+        new EventChannel('flutter.io/cameraPlugin/cameraEvents$_textureId')
```

Comment:
`plugins.flutter.io/cameraEvents` for consistency with other plugins.

Reply:
Done